src/libs/actions/IOU.ts

```diff
@@ -6337,6 +6338,96 @@ function approveMoneyRequest(expenseReport: OnyxEntry<OnyxTypes.Report>, full?:
     API.write(WRITE_COMMANDS.APPROVE_MONEY_REQUEST, parameters, {optimisticData, successData, failureData});
 }
 
+function unapproveExpenseReport(expenseReport: OnyxEntry<OnyxTypes.Report>) {
+    if (isEmptyObject(expenseReport)) {
+        return;
+    }
+
+    const currentNextStep = allNextSteps[`${ONYXKEYS.COLLECTION.NEXT_STEP}${expenseReport.reportID}`] ?? null;
+    const total = expenseReport.total ?? 0;
+
+    const optimisticUnapprovedReportAction = ReportUtils.buildOptimisticUnapprovedReportAction(total, expenseReport.currency ?? '', expenseReport.reportID);
```

Comment:
NAB - but looks like we don't need to extract to a `total` var?

```suggestion
    const optimisticUnapprovedReportAction = ReportUtils.buildOptimisticUnapprovedReportAction(expenseReport.total ?? 0, expenseReport.currency ?? '', expenseReport.reportID);
```

Reply:
my personal preference is to extract stuff even if it isn't being used.
it's easier to understand the code. 

eg: 
```
Onyx.merge(ONYX_KEYS.NEXT_STEP, allNextSteps[`${ONYXKEYS.COLLECTION.NEXT_STEP}${expenseReport.reportID}`] ?? null)
```

vs 

```

const nextStep = allNextSteps[`${ONYXKEYS.COLLECTION.NEXT_STEP}${expenseReport.reportID}`] ?? null

Onyx.merge(ONYX_KEYS.NEXT_STEP, nextStep)
```

Reply:
but in this case it's a simple value, so i'll remove the var

Reply:
Fair fair fair, and ya i agree in this case it's maybe not needed, but sometimes it can be very helpful